lib/servers/parameter_vault/parameter_vault.py

```diff
@@ -86,121 +86,131 @@ def save_parameters(self):
             fullDir = regDir + key
             yield self.client.registry.cd(fullDir)
             yield self.client.registry.set(parameter_name, value)
-    
+
     def _save_full(self, key, value):
-        t,item = self.parameters[key]
+        t, item = self.parameters[key]
         if t == 'parameter':
-            assert item[0] <= value <= item[1], "Parameter {} Out of Bound".format(key[1])
+            parameter_bound = "Parameter {} Out of Bound"
+            assert item[0] <= value <= item[1], parameter_bound.format(key[1])
             item[2] = value
             return (t, item)
         else:
             raise Exception("Can't save, not one of checkable types")
 
-    def check_parameter(self, name, value):
-        t,item = value
-        
-#        print name, t, t  == 'bool'
-        if t == 'parameter' or t == 'duration_bandwidth':
-            assert item[0] <= item[2] <= item[1], "Parameter {} Out of Bound".format(name)
+    def _check_parameter(self, key, value):
+        """
+        Parameters
+        ----------
+        key: str, parameter name
+        value:
+
+        Returns
+        -------
+        parameter "item" if parameter's value passes checks.
+        """
+        param_type, item = value
+
+        # Error strings
+        parameter_bound = "Parameter {} Out of Bound"
+        bad_selection = "Inorrect selection made in {}"
+
+        if param_type == 'parameter' or param_type == 'duration_bandwidth':
+            assert item[0] <= item[2] <= item[1], parameter_bound.format(key)
             return item[2]
-            
-        elif t == 'string' or t == 'bool' or t == 'sideband_selection' or t == 'spectrum_sensitivity':
+
+        elif param_type == 'string':
             return item
-            
-        elif t == 'scan':
-            minim,maxim = item[0]
-            start,stop,steps = item[1]
-            assert minim <= start <= maxim, "Parameter {} Out of Bound".format(name)
-            assert minim <= stop <= maxim, "Parameter {} Out of Bound".format(name)
+
+        elif param_type == 'bool':
+            return item
+
+        elif param_type == 'sideband_selection':
+            return item
+
+        elif param_type == 'spectrum_sensitivity':
+            return item
```

Comment:
maybe something like:

elif param_type in ["string", "bool", "sideband_selection", "spectrum_sensitivity"]:
    return item